plugins/state_history_plugin/state_history_plugin.cpp

```diff
@@ -366,6 +378,8 @@ void state_history_plugin::set_program_options(options_description& cli, options
            "enable debug mode for trace history");
    options("context-free-data-compression", bpo::value<string>()->default_value("zlib"), 
            "compression mode for context free data in transaction traces. Supported options are \"zlib\" and \"none\"");
+   options("state-history-allow-legacy-history-plugin", bpo::bool_switch()->default_value(false),
+           "allow legacy history plugin to run at the same time. Caution: it will cause random crashes");
```

Comment:
Why have this? I don't think this is needed. No one should run both.

Reply:
@heifner There are cases where you would want to run both for some amount of time to transition on a single node.  People have used it in this way and that's why we found out about the bug, so if they are okay with throwing caution to the wind then they can.  @linhuang-blockone I would state that it can cause random crashes, not that it will every time cases crashes.

Reply:
We have known issues history_plugin, it flat out does the wrong thing in a number of known cases. In my opinion we should kill it immediately even in 2.0.x.

Reply:
I agree.  I don't know why we still have this around and are enabling people to do the wrong thing.  No valid use case for running history + state_history in the same node at the same time.

Reply:
I will follow any decision once agreed upon.

FYI: This fix has already uncovered a number of tests which use both state_history plugin and directly or indirectly history plugin. I looked at one failed test: Ubuntu 18.04 ShiP test: it used history_api plugin (which requires history plugin) and state_history plugin.